Source/Core/Proj4Projection.js

```diff
@@ -0,0 +1,173 @@
+        this._forwardFailed = false;
+        this._inverseFailed = false;
+
+        heightScale = defaultValue(heightScale, 1.0);
+        this._heightScale = heightScale;
+        this._inverseHeightScale = 1.0 / heightScale;
+    }
+
+    defineProperties(Proj4Projection.prototype, {
+        /**
+         * The well-known-text string used to initialize proj4js.
+         * @memberof Proj4Projection.prototype
+         * @type {String}
+         * @readonly
+         */
+        wellKnownText: {
+            get: function() {
+                return this._wkt;
+            }
+        },
+        /**
+         * The scale for converting from heights in meters to the projection's units.
+         */
+        heightScale: {
+            get: function() {
+                return this._heightScale;
+            }
+        },
+        /**
+         * Gets whether or not the projection is cylindrical about the equator.
+         * Not all Proj4 projections are cylindrical about the equator.
+         *
+         * @memberof Proj4Projection.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         * @private
+         */
+        isEquatorialCylindrical : {
+            get : function() {
+                return false;
+            }
+        }
+    });
+
+    var projectionArray = [0, 0];
+    /**
+     * Projects a set of {@link Cartographic} coordinates, in radians, to map coordinates, in meters based on
+     * the specified projection.
+     *
+     * @param {Cartographic} cartographic The coordinates to project.
+     * @param {Cartesian3} [result] An instance into which to copy the result.  If this parameter is
+     *        undefined, a new instance is created and returned.
+     * @returns {Cartesian3} The projected coordinates.  If the result parameter is not undefined, the
+     *          coordinates are copied there and that instance is returned.  Otherwise, a new instance is
+     *          created and returned.
+     */
+    Proj4Projection.prototype.project = function(cartographic, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('cartographic', cartographic);
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new Cartesian3();
+        }
+
+        // without clamp proj4 might crash
+        projectionArray[0] = CesiumMath.clamp(CesiumMath.toDegrees(cartographic.longitude), -180 + CesiumMath.EPSILON7, 180 - CesiumMath.EPSILON7);
+        projectionArray[1] = CesiumMath.clamp(CesiumMath.toDegrees(cartographic.latitude), -90 + CesiumMath.EPSILON7, 90 - CesiumMath.EPSILON7);
+
+        var projected;
+        try {
+            projected = this._projection.forward(projectionArray);
+        } catch(e) {
+            if (!this._forwardFailed) {
+                // Log a warning the first time a projection fails
+                console.warn('proj4js forward failed for ' + projectionArray + ' with projection ' + this._wkt);
```

Comment:
Here and below, are we sure `console.warn` is the right approach compared to throwing an exception?  How often is `console.warn` used elsewhere in Cesium - and for what?

Reply:
Unfortunately Proj4js frequently fails when unprojecting positions in [EPSG:2039](http://spatialreference.org/ref/epsg/2039/), even when they appear to be in-bounds. It may also fail for other projections if camera navigation moves outside bounds, which we must assume will happen for Columbus View.

I'll change this to `oneTimeWarning`, which we use in other places.